PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/ControlScript/MasterControl/KeyboardMovement.rbxmx

```diff
@@ -73,54 +71,46 @@ function KeyboardMovement:Enable()
 		end
 	end
 	
-	local moveForwardFunc = function(actionName, inputState, inputObject)			
-		if inputState == Enum.UserInputState.Begin then
+	local moveForwardFunc = function(inputObject)		
+		if inputObject.UserInputState == Enum.UserInputState.Begin then
 			forwardValue = -1
-		elseif inputState == Enum.UserInputState.End then
+		elseif inputObject.UserInputState == Enum.UserInputState.End then
 			forwardValue = 0
 		end
 		updateMovement(inputState)
 	end
 	
-	local moveBackwardFunc = function(actionName, inputState, inputObject)	
-		if inputState == Enum.UserInputState.Begin then
+	local moveBackwardFunc = function(inputObject)	
+		if inputObject.UserInputState == Enum.UserInputState.Begin then
 			backwardValue = 1
-		elseif inputState == Enum.UserInputState.End then
+		elseif inputObject.UserInputState == Enum.UserInputState.End then
 			backwardValue = 0
 		end
 		updateMovement(inputState)
 	end
 	
-	local moveLeftFunc = function(actionName, inputState, inputObject)	
-		if inputState == Enum.UserInputState.Begin then
+	local moveLeftFunc = function(inputObject)
+		if inputObject.UserInputState == Enum.UserInputState.Begin then
 			leftValue = -1
-		elseif inputState == Enum.UserInputState.End then
+		elseif inputObject.UserInputState == Enum.UserInputState.End then
 			leftValue = 0
 		end
 		updateMovement(inputState)
 	end
 	
-	local moveRightFunc = function(actionName, inputState, inputObject)	
-		if inputState == Enum.UserInputState.Begin then
+	local moveRightFunc = function(inputObject)	
+		if inputObject.UserInputState == Enum.UserInputState.Begin then
 			rightValue = 1
-		elseif inputState == Enum.UserInputState.End then
+		elseif inputObject.UserInputState == Enum.UserInputState.End then
 			rightValue = 0
 		end
 		updateMovement(inputState)
 	end
-	
-	local jumpFunc = function(actionName, inputState, inputObject)
-		MasterControl:SetIsJumping(inputState == Enum.UserInputState.Begin)
+
+	local jumpFunc = function(inputObject)
+		MasterControl:SetIsJumping(inputObject.UserInputState == Enum.UserInputState.Begin)
 	end
 	
-	-- TODO: remove up and down arrows, these seem unnecessary
-	ContextActionService:BindActionToInputTypes("forwardMovement", moveForwardFunc, false, Enum.PlayerActions.CharacterForward)
-	ContextActionService:BindActionToInputTypes("backwardMovement", moveBackwardFunc, false, Enum.PlayerActions.CharacterBackward)
-	ContextActionService:BindActionToInputTypes("leftMovement", moveLeftFunc, false, Enum.PlayerActions.CharacterLeft)
-	ContextActionService:BindActionToInputTypes("rightMovement", moveRightFunc, false, Enum.PlayerActions.CharacterRight)
-	ContextActionService:BindActionToInputTypes("jumpAction", jumpFunc, false, Enum.PlayerActions.CharacterJump)
```

Comment:
I think we should just make these use ContextActionService:BindAction() (http://wiki.roblox.com/index.php?title=API:Class/ContextActionService/BindAction) instead of 
ContextActionService:BindActionToInputTypes